docs/topics/Permissions.md

```diff
@@ -163,19 +163,28 @@ Roles represent a set of permissions attached to a group of users. Roles have un
-| --------------- | --------- | ------------------------------------------------ |
-| id              | snowflake | role id                                          |
-| name            | string    | role name                                        |
-| color           | integer   | integer representation of hexadecimal color code |
-| hoist           | boolean   | if this role is pinned in the user listing       |
-| position        | integer   | position of this role                            |
-| permissions     | string    | permission bit set                               |
-| managed         | boolean   | whether this role is managed by an integration   |
-| mentionable     | boolean   | whether this role is mentionable                 |
+| Field       | Type                                                                         | Description                                      |
+| ----------- | ---------------------------------------------------------------------------- | ------------------------------------------------ |
+| id          | snowflake                                                                    | role id                                          |
+| name        | string                                                                       | role name                                        |
+| color       | integer                                                                      | integer representation of hexadecimal color code |
+| hoist       | boolean                                                                      | if this role is pinned in the user listing       |
+| position    | integer                                                                      | position of this role                            |
+| permissions | string                                                                       | permission bit set                               |
+| managed     | boolean                                                                      | whether this role is managed by an integration   |
+| mentionable | boolean                                                                      | whether this role is mentionable                 |
+| tags?       | [role tags](#DOCS_TOPICS_PERMISSIONS/role-object-role-tags-structure) object | the tags this role has                           |
 
 Roles without colors (`color == 0`) do not count towards the final computed color in the user list.
 
+###### Role Tags Structure
+
+| Field               | Type      | Description                                         |
+| ------------------- | --------- | --------------------------------------------------- |
+| bot_id?             | snowflake | the id of the bot this role belongs to              |
+| integration_id?     | snowflake | the id of the integration this role belongs to      |
+| premium_subscriber? | null      | whether this is the guild's premium subscriber role |
```

Comment:
What a weird way for us to pass this value. It's correct! Just weird, lol

Reply:
What does this actually mean? If the field is present, it should always be null...?

Reply:
If the field is present, it means the role is a premium subscriber role and the value is always null (if present)

Reply:
That's extremely strange - I'd love to hear what the thought behind that was, and what underlying type it actually has (assuming a nullable bool?)